modules/tracking/src/gtrTracker.cpp

```diff
@@ -54,7 +54,13 @@ void TrackerGOTURN::Params::write(cv::FileStorage& /*fs*/) const {}
 
 Ptr<TrackerGOTURN> TrackerGOTURN::createTracker(const TrackerGOTURN::Params &parameters)
 {
+#ifdef HAVE_OPENCV_DNN
     return Ptr<gtr::TrackerGOTURNImpl>(new gtr::TrackerGOTURNImpl(parameters));
+#else
+    (void)(parameters);
+    CV_Error(cv::Error::StsNotImplemented , "to use GOTURN, the tracking module needs to be built with opencv_dnn !");
+    return Ptr<TrackerGOTURN>();
```

Comment:
Is `TrackerGOTURN` abstract class? If so, it is better to use `CV_ErrorNoReturn` without "return" statement.